rest/server_context.go

```diff
@@ -1666,7 +1667,18 @@ func (sc *ServerContext) logStats(ctx context.Context) error {
 	// Marshal expvar map w/ timestamp to string and write to logs
 	base.RecordStats(string(marshalled))
 
-	return nil
+	if sc.Config.API.HeapProfileDisableCollection {
```

Comment:
The early returns here (1671 and 1677) feel out of place - it would be easy for someone to need to add something to logStats in future and just stick it at the end of the function.  Maybe move all of this into a sc.collectMemoryProfile() function to encapsulate this?